src/dependency-manager/src/utils/dictionary.ts

```diff
@@ -0,0 +1,3 @@
+export interface Dictionary<T> {
+  [key: string]: T;
```

Comment:
Should this be `[key: string]: T | undefined;`
Not sure if this is worth the hassle, but is technically correct for all random key lookups